base/action.yml

```
@@ -0,0 +1,26 @@
+# TODO: scripts下のパス決定を別ファイルに切り出し、それをここで呼び出すようにする。
+
+name: "Determine base"
+description: "プレビュー用ページのパスを決定する。"
+inputs:
+  name:
+    description: "preview-pages内での名前。"
+    required: true
+outputs:
+  base:
+    description: "ビルドしたプレビュー用ページのパス。"
+    value: ${{ steps.determine_base.outputs.base }}
+runs:
+  using: "composite"
+  steps:
+    - name: Determine base
+      uses: actions/github-script@v4
+      id: determine_base
+      with:
+        script: |
+          // ブランチの時はbranch-:name、
+          // PRの時はpr-:numberにビルドする。
```

（すごい細かいコメント）

`:name`が以外となにかわからないかもしれない。
（一瞬bashの初期値指定かと思った）

とはいえこれは解決策無さそう！
だけど日本語ならではの表現としてはこういうのもありかも！ダサいかもだけど！
```suggestion
          // ブランチの時は `branch-ブランチ名`、
          // PRの時は `pr-プルリクエスト番号` にビルドする。
```

まあ全然強くないコメントです！